### ISSUE TITLE
Unable to use into generated ***ClientProtocol into Test Target

### ISSUE THREAD

**Issue by adriantabirta** [grpc/grpc-swift#1526]
### Describe the bug

Got: Missing required module 'CNIOAtomics'

Hard: MacBook M1
Soft: latest Xcode version on Ventura, swift version 5.7.1, grpc-swift(1.10.0) in SPM package,

SPM package [here](https://github.com/adriantabirta/taxify-networking)

### To reproduce

Steps to reproduce the bug you've found:
1. Create simple Xcode project and add package as dependency 
2. Try to use **GeneratedServiceClientProtocol** to create **RideServiceClientProtocolMock** into test target
3. Got error/s

### Expected behaviour

I want to use generated client protocol to create mock classes for testing.

### Additional information

- Have tried to use `@_exported import ...` in SPM package - same error,
- Have tried to add grpc-swift as dependency into Xcode project alongside with created SPM package - error, duplicated code
- Added created package to `Link binary with Libraries` in Test Target - give me errors like:
```
Undefined symbols for architecture arm64:
  "_$s19TaxifyClientSwiftUI20RideRemoteDataSourceMp", referenced from:
      _symbolic ______pSg 19TaxifyClientSwiftUI20RideRemoteDataSourceP in RealRideRemoteDataSourceTests.o
```

Is there a way to forward correctly dependencies like `grpc-swift -> my SPM package -> Xcode (especially test target)`?
Notice: Mocking is working on application target, am I doing something wrong?

**Comment by Lukasa** [comment]
This seems like an Xcode issue: does this build correctly otherwise?

**Comment by adriantabirta** [comment]
> This seems like an Xcode issue: does this build correctly otherwise?

otherwise how?

**Comment by Lukasa** [comment]
Yeah, I think this will require filing a report using Feedback Assistant or asking for help on the Apple Developer Forums. This issue appears to be with Xcode, not with grpc-swift.
